server/models/Expense.js

```diff
@@ -531,28 +538,25 @@ function defineModel() {
       INNER JOIN "Transactions" t
         ON t."ExpenseId" = e."id"
       INNER JOIN "Collectives" c
-        ON (
-          c."id" = $collectiveId
-          ${includeChildren ? `OR c."ParentCollectiveId" = $collectiveId` : ``}
-        )
-        AND c."deletedAt" IS NULL
+        ON c."id" = t."CollectiveId" AND c."deletedAt" IS NULL
```

Comment:
I don't get the reason for this change; unless it was buggy, it seems less verbose & more performant to me to have a simple SQL `id = :id OR ParentCollectiveId = :id` than having 2 queries to get children then filter.

Reply:
It's personal preference, the idea is to simplify the query by not mixing filtering logic with the join operation.
TBH, I'm not sure how this will affect performance, but I can analyze it further if necessary.

Reply:
Filtering in `JOIN` is fine IMO, unless I'm missing something I don't think it's an anti-pattern. We can move the condition to `WHERE` if it's clearer.

Altogether I don't think this refactor adds any clarity; I find it more difficult to read a 7 lines JS filter (that gets duplicated in another function below) than a single-line SQL filter. I don't think there's a massive performance difference, but it's almost certain that these 2 queries will be slower than a single (supposedly well-indexed) one. Also, "No need to touch it if it works" :smile: